vault/blog.2021.2021-08-30-highlights-vscode-159.md

```diff
@@ -0,0 +1,66 @@
+
+VSCode users are often bouncing around different files in their workspaces. By default, if you close out a file tab, VSCode doesn't track where you were in that file. This can be annoying if you didn't mean to close out the file.
+
+Say hello to shared view states across editor groups[^4]. Editor groups are the windows of tab collections within the VSCode editor[^5] that grow whenever **Split Editor/Open to the Side** commands are used. These groups can remember your past so that your rampant mind doesn't need to.
+
+![GIF showing the share view state feature with The House on The Borderland, by William Hope Hodgson](https://org-dendron-public-assets.s3.amazonaws.com/images/vscode-share-view-state.gif)
+
+> Ah yes, that's right, I left off at the pig people.[^6]
+
+### Terminal magic tricks
+
+By default, a new terminal can be created with `Ctrl+Tilde(~)`, where it pops up in the bottom as an entryway into the powerful, dark underbelly of shells. VSCode 1.59 highlights some wizardry alternatives.
+
+#### Create terminals to the side
+
+Are you one of those people who spins a monitor into portrait mode, because of some insatiable love for 90<sup>o</sup> angles? Maybe you simply prefer tall terminals? `Ctrl+Shift+P` and `Terminal: Create Terminal in Editor Area to the Side`[^7], will be exactly what you're looking for.
+
+![GIF showing the creation of terminals to the side, instead of defaulting to the bottom](https://org-dendron-public-assets.s3.amazonaws.com/images/vscode-side-terminals.gif)
+
+#### Drag and drop terminals across windows
+
+How many VSCode windows are sitting open on your desktop? Are you taking notes in one workspace, while continuing your commands in a separate VSCode window? Make your life easier, retain all your in-shell variables, and drag the active terminal from one window to another.
+
+![GIF showing the drag-and-drop feature of terminals into different VSCode windows](https://org-dendron-public-assets.s3.amazonaws.com/images/vscode-drag-drop-terminals.gif)
+
+This can be helpful when improving the documentation in one window as active development is happening in another.
+
+### Other notables
+
+> General tip: Wondering what shortcuts are configured in VSCode? One shortcut to rule them all: `Ctrl+K Ctrl+S`
+- Or `Ctrl+Shift+P` -> **Preferences: Open Keyboard Shortcuts**
+
+- If you're using an M1 Mac, you should get the latest VSCode 1.59.1[^8] point release, as it addresses a bug where VSCode may crash.
+- A preview feature has been added for automatic language detection in unsaved/untitled files. `experimentalLanguageDetection`[^9] (search against settings via `Ctrl+Comma`) can be set to `true`, which uses an ML model from `yoeo/Guesslang`[^10] with the open-source ML library `Tensorflow.js`[^11]. VSCode is planning for this feature to be `true` by default in the future, so that code is recognized out-of-box, instead of always relying on the file extension or force **Change Language Mode** (`Ctrl+K M`).
```

Comment:
link to docs about what *preview feature* means in vscode

Reply:
Extra note included in footnotes about what these _preview features_ are. They seemed to have left out the little note they usually include in their release notes about it